packages/core/core/src/BundleGraph.js

```diff
@@ -450,13 +454,29 @@ export default class BundleGraph {
     return {asset, exportSymbol: symbol, symbol: identifier};
   }
 
-  getHash(bundle: Bundle): string {
+  _getContentHash(bundle: Bundle): string {
+    let existingHash = this._bundleContentHashes.get(bundle.id);
+    if (existingHash != null) {
+      return existingHash;
+    }
+
     let hash = crypto.createHash('md5');
     // TODO: sort??
     this.traverseAssets(bundle, asset => {
       hash.update(asset.outputHash);
     });
 
+    let hashHex = hash.digest('hex');
+    this._bundleContentHashes.set(bundle.id, hashHex);
```

Comment:
Since computing the hash for a bundle how involves computing content hashes for its entire bundle subgraph (in cases where there's only one entry, this is every bundle), cache them.